File: backend/app/schemas/order.py
import uuid
from pydantic import BaseModel
from typing import List, Optional, Dict, Any
from decimal import Decimal
from .product import ProductVariant

class OrderItem(BaseModel):
    id: uuid.UUID
    product_variant_id: uuid.UUID
    quantity: int
    unit_price: Decimal
    product_variant: ProductVariant

    class Config:
        orm_mode = True

class Order(BaseModel):
    id: uuid.UUID
    order_number: str
    status: str
    total_amount: Decimal
    shipping_address: Dict[str, Any]
    items: List[OrderItem]

    class Config:
        orm_mode = True

class CheckoutSessionCreate(BaseModel):
    shipping_address: Dict[str, Any]

class CheckoutSession(BaseModel):
    client_secret: str
    order_id: uuid.UUID

File: frontend/src/types/order.ts
import type { ProductVariant } from './product';

export interface OrderItem {
    id: string;
    product_variant_id: string;
    quantity: number;
    unit_price: number;
    product_variant: ProductVariant;
}

export interface Order {
    id: string;
    order_number: string;
    status: string;
    total_amount: number;
    shipping_address: { [key: string]: any };
    items: OrderItem[];
    created_at: string; // Assuming this field exists from the database model
}
